AzureAttestationExtension/agent.py

```diff
@@ -0,0 +1,708 @@
+    """
+    Log error message, being cautious of possibility that waagent may not be
+    imported
+    """
+    if 'Utils.WAAgentUtil' in sys.modules:
+        waagent.Error(message)
+    else:
+        print('Error: {0}'.format(message))
+
+
+def hutil_log_info(message):
+    """
+    Log informational message, being cautious of possibility that hutil may
+    not be imported and configured
+    """
+    if HUtilObject is not None:
+        HUtilObject.log(message)
+    else:
+        print('Info: {0}'.format(message))
+
+
+def hutil_log_error(message):
+    """
+    Log error message, being cautious of possibility that hutil may not be
+    imported and configured
+    """
+    if HUtilObject is not None:
+        HUtilObject.error(message)
+    else:
+        print('Error: {0}'.format(message))
+
+
+def log_and_exit(operation, exit_code = 1, message = ''):
+    """
+    Log the exit message and perform the exit
+    """
+    if exit_code is 0:
+        waagent_log_info(message)
+        hutil_log_info(message)
+        exit_status = 'success'
+    else:
+        waagent_log_error(message)
+        hutil_log_error(message)
+        exit_status = 'failed'
+
+    if HUtilObject is not None:
+        HUtilObject.do_exit(exit_code, operation, exit_status, str(exit_code),
+                            message)
+    else:
+        update_status_file(operation, str(exit_code), exit_status, message)
+        sys.exit(exit_code)
+
+def get_log_directory():
+    if HUtilObject is None:
+        handler_env = get_handler_env()
+        return str(handler_env['handlerEnvironment']['logFolder'])
+    else:
+        return HUtilObject._context._log_dir
+
+def start_attestation_client():
+    """
+    Start attestation process that performs periodic monitoring of attestion process.
+    :return: None
+
+    """
+    stop_process('attestation_client')    
+    log_dir = get_log_directory()
+    attestation_client_filepath = os.path.join(os.getcwd(),'attestationclient')
+    args = [attestation_client_filepath, '-e', MaaEndpoint, '-a' AscReportingEndpoint, '-l' log_dir ]
+    log = open(os.path.join(os.getcwd(), 'daemon.log'), 'w')
+    
+    child = subprocess.Popen(args, stdout=log, stderr=log)        
+    pids_filepath = os.path.join(os.getcwd(), 'attestation_client.pid')
+    
+    with open(pids_filepath, 'w') as f:
+        f.write(str(child.pid) + '\n')
+  
+def stop_attestation_client(processname):
+    pid_file = 'attestation_client.pid'
+    pids_filepath = os.path.join(os.getcwd(),pid_file)
+
+    # kill existing attestation process
+    if os.path.exists(pids_filepath):
+        with open(pids_filepath, "r") as f:
+            for pids in f.readlines():
+                kill_cmd = "kill " + pids
+                run_command_and_log(kill_cmd)
+                run_command_and_log("rm "+pids_filepath)
+
+def stop_attestation_client_process():    
+    stop_process('attestation_extension')
+
+    return 0, "Successfully stopped attestation proces"
```

Comment:
I couldn't find the definition of the stop_Process, hence not sure about the behavior of error case handling. I still feel we are returning 0 without checking the return code of stop_process.

Reply:
will fix it.

Reply:
Also small typo in logging: **proces**